### PR TITLE
Handle expressions that result in top-level case statements in Search…

### PR DESCRIPTION
**Pull request by tuespetre**
…ConditionTranslatingVisitor

Addresses #7271

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs [hunk 1]
```diff
@@ -1652,68 +1652,47 @@ public static bool IsSearchCondition(Expression expression)
 
             protected override Expression VisitBinary(BinaryExpression expression)
             {
-                Expression newLeft;
-                Expression newRight;
+                var parentIsSearchCondition = _isSearchCondition;
 
-                if (_isSearchCondition)
+                if (expression.IsLogicalOperation())
```
Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs]: simplify this as `_isSearchCondition = expression.IsLogicalOperation()`

File: src/Microsoft.EntityFrameworkCore.Specification.Tests/QueryTestBase.cs [hunk 2]
```diff
@@ -2029,6 +2029,43 @@ public virtual void Where_Is_on_same_type()
...
+        }
+
+        [ConditionalFact]
+        public virtual void Where_binary_or_with_binary_or()
+        {
+            AssertQuery<Order>(os => os
+                .Where(o => o.OrderID > 54321 | o.OrderID > 98765 | o.OrderID > 1234567));
+        }
+
+        [ConditionalFact]
```
Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore.Specification.Tests/QueryTestBase.cs]: Just for coverage - Test for binary and with binary and.

File: src/Microsoft.EntityFrameworkCore.Specification.Tests/QueryTestBase.cs [hunk 3]
```diff
@@ -2029,6 +2029,43 @@ public virtual void Where_Is_on_same_type()
         }
 
         [ConditionalFact]
```
Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore.Specification.Tests/QueryTestBase.cs]: Also add same tests in `AsyncQueryTestBase`. You don't need to assert SQL gen for those.

**Comment by smitpatel** [comment]
@tuespetre - Please update PR with requested changes and rebase it on latest dev.

**Comment by rowanmiller** [comment]
@tuespetre are you planning to address the feedback on this PR?

**Comment by tuespetre** [comment]
Oh, sorry -- I forgot to reply. Yes, I am planning to.

**Comment by tuespetre** [comment]
@smitpatel looks like this is obsoleted by your new PR, want me to close this one?

**Comment by smitpatel** [comment]
@tuespetre - Thank you for your efforts. We had been pending doing a refactor in that part of query pipeline to improve accuracy since we had seen good incoming traffic of bugs in that area. Due to being closer to correct solution, the underlying issue was getting resolved in refactoring. Since the refactored code was different approach from the one outlined in this PR (which is along the line of older code), we couldn't take this in as refactoring would be changing all of it. Though thank you for your help.

**Comment by tuespetre** [comment]
Anytime!

**Pull request closed without merging**